modular_skyrat/modules/interaction_menu/code/interaction_component.dm

```diff
@@ -113,4 +143,85 @@
 			interact_next = interaction_component.interact_last + INTERACTION_COOLDOWN
 			interaction_component.interact_next = interact_next
 			return TRUE
+
+	if(params["item_slot"])
+		// This code should be easy enough to follow... I hope.
+		var/item_index = params["item_slot"]
+		var/mob/living/carbon/human/source = locate(params["userref"])
+		var/mob/living/carbon/human/target = locate(params["selfref"])
+		var/obj/item/new_item = source.get_active_held_item()
+		var/obj/item/existing_item = target.vars[item_index]
+
+		if(!existing_item && !new_item)
+			source.show_message(span_warning("No item to insert or remove!"))
+			return
+
+		if(!existing_item && !istype(new_item, /obj/item/clothing/sextoy))
+			source.show_message(span_warning("The item you're holding is not a toy!"))
+			return
+
+		if(can_lewd_strip(source, target, item_index) && is_toy_compatible(new_item, item_index))
+			var/internal = (item_index in list(ORGAN_SLOT_VAGINA, ORGAN_SLOT_ANUS))
+			var/insert_or_attach = internal ? "insert" : "attach"
+			var/into_or_onto = internal ? "into" : "onto"
+
+			if(existing_item)
+				source.visible_message(span_purple("[source.name] starts trying to remove something from [target.name]'s [item_index]."), span_purple("You start to remove [existing_item.name] from [target.name]'s [item_index]."), span_purple("You hear someone trying to remove something from someone nearby."), vision_distance = 1, ignored_mobs = list(target))
+			else if (new_item)
+				source.visible_message(span_purple("[source.name] starts trying to [insert_or_attach] the [new_item.name] [into_or_onto] [target.name]'s [item_index]."), span_purple("You start to [insert_or_attach] the [new_item.name] [into_or_onto] [target.name]'s [item_index]."), span_purple("You hear someone trying to [insert_or_attach] something [into_or_onto] someone nearby."), vision_distance = 1, ignored_mobs = list(target))
+			if (source != target)
+				target.show_message(span_warning("[source.name] is trying to [existing_item ? "remove the [existing_item.name] [internal ? "in" : "on"]" : new_item ? "is trying to [insert_or_attach] the [new_item.name] [into_or_onto]" : span_alert("What the fuck, impossible condition? interaction_component.dm!")] your [item_index]!"))
+			if(do_after(
+				source,
+				5 SECONDS,
+				target,
+				interaction_key = "interation_[item_index]"
+				) && can_lewd_strip(source, target, item_index))
```

Comment:
```suggestion
			if(
				do_after(source, 5 SECONDS, target, interaction_key = "interation_[item_index]")
				&& can_lewd_strip(source, target, item_index)
			)
```

Reply:
Nope, byond must have `&&` after a condition on the same line. I don't make the rules, DM does.